### PR TITLE
chore: bump and refactor images, code cleanup

### PR DESCRIPTION
**Pull request by jonaro00**
## Description of change

- Bump stack's rust version to 1.73, CI and MSRV remains 1.72
- Some refactorings to allow use of smaller images
- Copy executable instead of move for faster redeploys

## How has this been tested? (if applicable)

stack went up locally

### CODE REVIEW

File: Containerfile [hunk 1]
```diff
@@ -54,14 +54,16 @@ RUN cargo build \
     --bin shuttle-next -F next
 
 
+####### Helper step
+
+FROM docker.io/library/debian:bookworm-20230904-slim AS bookworm-20230904-slim-plus
+RUN apt update && apt install -y curl ca-certificates; rm -rf /var/lib/apt/lists/*
```
Comment (chesedo) [inline Containerfile:60]: Why are we doing this for all projects and not just `auth`?

File: runtime/Cargo.toml [hunk 2]
```diff
@@ -27,7 +27,7 @@ tokio = { workspace = true, features = ["full"] }
 tokio-stream = { workspace = true }
 tonic = { workspace = true }
 tower = { workspace = true }
-cap-std = { workspace = true, optional = true }
+cap-std = { version = "1.0.15", optional = true }
```
Comment (chesedo) [inline runtime/Cargo.toml]: Why the downgrade from v2 (in the workspace)?

**Review by chesedo — COMMENTED** [review]
Just one comment

File: runtime/Cargo.toml (hunk 2 above)
Comment (chesedo) [reply runtime/Cargo.toml]: Nvm, figured this out

File: Containerfile (hunk 1 above)
Comment (jonaro00) [reply Containerfile:60]: Common for auth, gw, and prov was that they can run in the slim image only if they have ca-certs and/or curl. Therefore, this middle step was made.

Comment (chesedo) [reply Containerfile:60]: Ohh, I see 👍

**Review by chesedo — APPROVED** [review]
LGTM

**Pull request merged**